From 330a1245cb91583d9bc916bbb6d8c7d2c86b26f3 Mon Sep 17 00:00:00 2001
From: Fugang Duan <b38611@freescale.com>
Date: Wed, 4 Mar 2015 15:26:59 +0800
Subject: [PATCH] MLK-10367 tty: serial: imx: set uart module clock to 80Mhz

When UART module clock is great than 80Mhz, there may have risk after
confirming with IC owner. So set the maximum module clock to 80Mhz.

Signed-off-by: Fugang Duan <B38611@freescale.com>
---
 drivers/tty/serial/imx.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/tty/serial/imx.c b/drivers/tty/serial/imx.c
index 4a5c69c..93a0c65 100644
--- a/drivers/tty/serial/imx.c
+++ b/drivers/tty/serial/imx.c
@@ -188,6 +188,7 @@
 
 #define UART_NR 8
 #define IMX_RXBD_NUM 20
+#define IMX_MODULE_MAX_CLK_RATE	80000000
 
 /* i.mx21 type uart runs on all i.mx except i.mx1 */
 enum imx_uart_type {
@@ -2107,6 +2108,8 @@ static int serial_imx_probe(struct platform_device *pdev)
 		return ret;
 	}
 
+	if (sport->clk_per > IMX_MODULE_MAX_CLK_RATE)
+		clk_set_rate(sport->clk_per, IMX_MODULE_MAX_CLK_RATE);
 	sport->port.uartclk = clk_get_rate(sport->clk_per);
 
 	imx_ports[sport->port.line] = sport;
-- 
1.8.0

